build/cmake/SGX.cmake

```diff
@@ -0,0 +1,243 @@
+# Copyright 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+################################################################################
+# Environment Variables
+################################################################################
+
+IF (NOT DEFINED ENV{PDO_ENCLAVE_CODE_SIGN_PEM})
+  MESSAGE(FATAL_ERROR "PDO_ENCLAVE_CODE_SIGN_PEM not defined")
+ENDIF()
+SET(PDO_ENCLAVE_CODE_SIGN_PEM "$ENV{PDO_ENCLAVE_CODE_SIGN_PEM}")
+
+IF (NOT DEFINED ENV{SGX_MODE})
+  MESSAGE(FATAL_ERROR "SGX_MODE not defined")
+ENDIF()
+SET(SGX_MODE $ENV{SGX_MODE})
+
+IF (${SGX_MODE} STREQUAL "SIM")
+    SET(SGX_USE_SIMULATOR TRUE)
+ELSE()
+    SET(SGX_USE_SIMULATOR FALSE)
+ENDIF()
+
+IF (NOT DEFINED ENV{SGX_SDK})
+  MESSAGE(FATAL_ERROR "SGX_SDK not defined")
+ENDIF()
+SET(SGX_SDK $ENV{SGX_SDK})
+
+IF (NOT DEFINED ENV{SGX_SSL})
+  MESSAGE(FATAL_ERROR "SGX_SSL not defined")
+ENDIF()
+SET(SGX_SSL $ENV{SGX_SSL})
+
+# The expectation is that these are set prior to including this file
+IF (NOT DEFINED CMAKE_LIBRARY_OUTPUT_DIRECTORY)
+  MESSAGE(FATAL_ERROR "CMAKE_LIBRARY_OUTPUTDIRECTORY must be set")
+ENDIF()
+
+################################################################################
+# Internal SGX Variables
+################################################################################
+SET(SGX_EDGER "${SGX_SDK}/bin/x64/sgx_edger8r")
+SET(SGX_SIGN "${SGX_SDK}/bin/x64/sgx_sign")
+
+IF (${SGX_USE_SIMULATOR})
+  ADD_COMPILE_DEFINITIONS(SGX_SIMULATOR=1)
+  SET(TRTS_LIBRARY_NAME "sgx_trts_sim")
+  SET(URTS_LIBRARY_NAME "sgx_urts_sim")
+  SET(AE_SERVICE_LIBRARY_NAME "sgx_uae_service_sim")
+  SET(SERVICE_LIBRARY_NAME "sgx_tservice_sim")
+ELSE()
+  SET(TRTS_LIBRARY_NAME "sgx_trts")
+  SET(URTS_LIBRARY_NAME "sgx_urts")
+  SET(AE_SERVICE_LIBRARY_NAME "")
+  SET(SERVICE_LIBRARY_NAME "sgx_tservice")
+ENDIF()
+
+SET(SGX_TRUSTED_LIBS sgx_tstdc sgx_tcxx sgx_tcrypto ${SERVICE_LIBRARY_NAME})
+SET(SGX_UNTRUSTED_LIBS ${URTS_LIBRARY_NAME} ${AE_SERVICE_NAME} pthread)
```

Comment:
Being deprecated, I believe `sgx_uae_service` is now functionally empty. So it's not necessary to add it.

Also, the `AE_SERVICE_NAME` variable seems always.